### ISSUE TITLE
npm install react-admin upstream dependency conflict

### ISSUE THREAD

**Issue by BitBoxBear** [marmelab/react-admin#5908]
node -v: 15.8.0
npm -v: 7.5.1
OS: Windows 10 64

command: `npm install react-admin`

0 verbose cli [
0 verbose cli   'C:\\NodeJS\\node.exe',
0 verbose cli   'C:\\NodeJS\\node_modules\\npm\\bin\\npm-cli.js',
0 verbose cli   'install',
0 verbose cli   'react-admin'
0 verbose cli ]
1 info using npm@7.5.1
2 info using node@v15.8.0
3 timing config:load:defaults Completed in 1ms
4 timing config:load:file:D:\NodeJS\node_modules\npm\npmrc Completed in 2ms
5 timing config:load:builtin Completed in 2ms
6 timing config:load:cli Completed in 1ms
7 timing config:load:env Completed in 0ms
8 timing config:load:file:E:\front-end-admin\.npmrc Completed in 0ms
9 timing config:load:project Completed in 2ms
10 timing config:load:file:C:\Users\X\.npmrc Completed in 0ms
11 timing config:load:user Completed in 0ms
12 timing config:load:file:C:\Users\X\AppData\Roaming\npm\etc\npmrc Completed in 0ms
13 timing config:load:global Completed in 0ms
14 timing config:load:cafile Completed in 0ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:setUserAgent Completed in 1ms
17 timing config:load:setEnvs Completed in 0ms
18 timing config:load Completed in 7ms
19 verbose npm-session 3e4d2d80daee4988
20 timing npm:load Completed in 13ms
21 timing arborist:ctor Completed in 1ms
22 timing arborist:ctor Completed in 0ms
23 timing idealTree:init Completed in 9ms
24 timing idealTree:userRequests Completed in 1ms
25 silly idealTree buildDeps
26 silly fetch manifest react-admin@*
27 http fetch GET 304 https://registry.npmjs.org/react-admin 1008ms (from cache)
28 silly fetch manifest react@^16.9.0 || ^17.0.0
29 http fetch GET 304 https://registry.npmjs.org/react 66ms (from cache)
30 silly fetch manifest react-dom@^16.9.0 || ^17.0.0
31 http fetch GET 304 https://registry.npmjs.org/react-dom 67ms (from cache)
32 silly placeDep ROOT react-admin@3.12.2 OK for:  want: *
33 silly placeDep ROOT react@17.0.1 OK for: react-admin@3.12.2 want: ^16.9.0 || ^17.0.0
34 silly placeDep ROOT react-dom@17.0.1 OK for: react-admin@3.12.2 want: ^16.9.0 || ^17.0.0
35 silly fetch manifest @material-ui/core@^4.11.2
36 silly fetch manifest @material-ui/icons@^4.11.2
37 silly fetch manifest @material-ui/styles@^4.11.2
38 silly fetch manifest connected-react-router@^6.5.2
39 silly fetch manifest final-form@^4.20.0
40 silly fetch manifest final-form-arrays@^3.0.1
41 silly fetch manifest ra-core@^3.12.2
42 silly fetch manifest ra-i18n-polyglot@^3.12.2
43 silly fetch manifest ra-language-english@^3.12.2
44 silly fetch manifest ra-ui-materialui@^3.12.2
45 silly fetch manifest react-final-form@^6.3.3
46 silly fetch manifest react-final-form-arrays@^3.1.1
47 silly fetch manifest react-redux@^7.1.0
48 silly fetch manifest react-router@^5.1.0
49 silly fetch manifest react-router-dom@^5.1.0
50 silly fetch manifest redux@^3.7.2 || ^4.0.3
51 silly fetch manifest redux-saga@^1.0.0
52 silly fetch manifest loose-envify@^1.1.0
53 silly fetch manifest object-assign@^4.1.1
54 silly fetch manifest scheduler@^0.20.1
55 http fetch GET 304 https://registry.npmjs.org/ra-core 154ms (from cache)
56 http fetch GET 304 https://registry.npmjs.org/ra-i18n-polyglot 170ms (from cache)
57 http fetch GET 304 https://registry.npmjs.org/ra-language-english 175ms (from cache)
58 http fetch GET 304 https://registry.npmjs.org/react-final-form 179ms (from cache)
59 http fetch GET 304 https://registry.npmjs.org/final-form-arrays 186ms (from cache)
60 http fetch GET 304 https://registry.npmjs.org/final-form 186ms (from cache)
61 http fetch GET 304 https://registry.npmjs.org/ra-ui-materialui 184ms (from cache)
62 http fetch GET 304 https://registry.npmjs.org/react-final-form-arrays 192ms (from cache)
63 http fetch GET 304 https://registry.npmjs.org/@material-ui%2ficons 202ms (from cache)
64 http fetch GET 304 https://registry.npmjs.org/connected-react-router 201ms (from cache)
65 http fetch GET 304 https://registry.npmjs.org/react-router 204ms (from cache)
66 http fetch GET 304 https://registry.npmjs.org/redux 204ms (from cache)
67 http fetch GET 304 https://registry.npmjs.org/react-redux 208ms (from cache)
68 http fetch GET 304 https://registry.npmjs.org/object-assign 208ms (from cache)
69 http fetch GET 304 https://registry.npmjs.org/loose-envify 217ms (from cache)
70 http fetch GET 304 https://registry.npmjs.org/react-router-dom 219ms (from cache)
71 http fetch GET 304 https://registry.npmjs.org/scheduler 215ms (from cache)
72 http fetch GET 304 https://registry.npmjs.org/redux-saga 225ms (from cache)
73 http fetch GET 304 https://registry.npmjs.org/@material-ui%2fcore 281ms (from cache)
74 http fetch GET 304 https://registry.npmjs.org/@material-ui%2fstyles 294ms (from cache)
75 timing idealTree:#root Completed in 1452ms
76 silly placeDep ROOT loose-envify@1.4.0 OK for: react@17.0.1 want: ^1.1.0
77 silly placeDep ROOT object-assign@4.1.1 OK for: react@17.0.1 want: ^4.1.1
78 silly fetch manifest js-tokens@^3.0.0 || ^4.0.0
79 http fetch GET 304 https://registry.npmjs.org/js-tokens 50ms (from cache)
80 timing idealTree:node_modules/react Completed in 53ms
81 silly placeDep ROOT js-tokens@4.0.0 OK for: loose-envify@1.4.0 want: ^3.0.0 || ^4.0.0
82 timing idealTree:node_modules/loose-envify Completed in 1ms
83 timing idealTree:node_modules/js-tokens Completed in 0ms
84 timing idealTree:node_modules/object-assign Completed in 0ms
85 silly fetch manifest @types/react@^16.8.6 || ^17.0.0
86 http fetch GET 304 https://registry.npmjs.org/@types%2freact 130ms (from cache)
87 silly fetch manifest history@^4.7.2
88 http fetch GET 304 https://registry.npmjs.org/history 52ms (from cache)
89 silly fetch manifest immutable@^3.8.1 || ^4.0.0-rc.1
90 http fetch GET 304 https://registry.npmjs.org/immutable 55ms (from cache)
91 silly fetch manifest react@^16.4.0
92 silly fetch manifest seamless-immutable@^7.1.3
93 http fetch GET 304 https://registry.npmjs.org/seamless-immutable 52ms (from cache)
94 silly placeDep ROOT @material-ui/core@4.11.3 OK for: react-admin@3.12.2 want: ^4.11.2
95 silly placeDep ROOT @material-ui/icons@4.11.2 OK for: react-admin@3.12.2 want: ^4.11.2
96 silly placeDep ROOT @material-ui/styles@4.11.3 OK for: react-admin@3.12.2 want: ^4.11.2
97 silly placeDep node_modules/react-admin connected-react-router@6.8.0 OK for: react-admin@3.12.2 want: ^6.5.2
98 silly placeDep ROOT immutable@4.0.0-rc.12 OK for: connected-react-router@6.8.0 want: ^3.8.1 || ^4.0.0-rc.1
99 silly placeDep ROOT history@4.10.1 OK for: connected-react-router@6.8.0 want: ^4.7.2
100 timing idealTree Completed in 1846ms
101 timing command:install Completed in 1852ms
102 verbose stack Error: unable to resolve dependency tree
102 verbose stack     at Arborist.[failPeerConflict] (D:\NodeJS\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\build-ideal-tree.js:1115:25)
102 verbose stack     at Arborist.[placeDep] (D:\NodeJS\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\build-ideal-tree.js:1201:30)
102 verbose stack     at Arborist.[placeDep] (D:\NodeJS\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\build-ideal-tree.js:1339:41)
102 verbose stack     at D:\NodeJS\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\build-ideal-tree.js:835:46
102 verbose stack     at Array.map (<anonymous>)
102 verbose stack     at Arborist.[buildDepStep] (D:\NodeJS\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\build-ideal-tree.js:835:8)
102 verbose stack     at async Arborist.buildIdealTree (D:\NodeJS\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\build-ideal-tree.js:209:7)
102 verbose stack     at async Promise.all (index 1)
102 verbose stack     at async Arborist.reify (D:\NodeJS\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\reify.js:127:5)
102 verbose stack     at async install (D:\NodeJS\node_modules\npm\lib\install.js:38:3)
103 verbose cwd E:\front-end-admin
104 verbose Windows_NT 10.0.19042
105 verbose argv "C:\\NodeJS\\node.exe" "C:\\NodeJS\\node_modules\\npm\\bin\\npm-cli.js" "install" "react-admin"
106 verbose node v15.8.0
107 verbose npm  v7.5.1
108 error code ERESOLVE
109 error ERESOLVE unable to resolve dependency tree
110 error
111 error Found: [1mreact[22m@[1m17.0.1[22m[2m[22m
111 error [2mnode_modules/react[22m
111 error   [35mpeer[39m [1mreact[22m@"[1m^16.9.0 || ^17.0.0[22m" from [1mreact-admin[22m@[1m3.12.2[22m[2m[22m
111 error   [2mnode_modules/react-admin[22m
111 error     [1mreact-admin[22m@"[1m*[22m" from the root project
111 error   [35mpeer[39m [1mreact[22m@"[1m17.0.1[22m" from [1mreact-dom[22m@[1m17.0.1[22m[2m[22m
111 error   [2mnode_modules/react-dom[22m
111 error     [35mpeer[39m [1mreact-dom[22m@"[1m^16.9.0 || ^17.0.0[22m" from [1mreact-admin[22m@[1m3.12.2[22m[2m[22m
111 error     [2mnode_modules/react-admin[22m
111 error       [1mreact-admin[22m@"[1m*[22m" from the root project
111 error     [35mpeer[39m [1mreact-dom[22m@"[1m^16.8.0 || ^17.0.0[22m" from [1m@material-ui/core[22m@[1m4.11.3[22m[2m[22m
111 error     [2mnode_modules/@material-ui/core[22m
111 error       [1m@material-ui/core[22m@"[1m^4.11.2[22m" from [1mreact-admin[22m@[1m3.12.2[22m[2m[22m
111 error       [2mnode_modules/react-admin[22m
111 error         [1mreact-admin[22m@"[1m*[22m" from the root project
111 error       1 more (@material-ui/icons)
111 error     2 more (@material-ui/icons, @material-ui/styles)
111 error   3 more (@material-ui/core, @material-ui/icons, @material-ui/styles)
111 error
111 error Could not resolve dependency:
111 error [35mpeer[39m [1mreact[22m@"[1m^16.4.0[22m" from [1mconnected-react-router[22m@[1m6.8.0[22m[2m[22m
111 error [2mnode_modules/react-admin/node_modules/connected-react-router[22m
111 error   [1mconnected-react-router[22m@"[1m^6.5.2[22m" from [1mreact-admin[22m@[1m3.12.2[22m[2m[22m
111 error   [2mnode_modules/react-admin[22m
111 error     [1mreact-admin[22m@"[1m*[22m" from the root project
111 error
111 error Fix the upstream dependency conflict, or retry
111 error this command with --force, or --legacy-peer-deps
111 error to accept an incorrect (and potentially broken) dependency resolution.
111 error
111 error See C:\Users\X\AppData\Local\npm-cache\eresolve-report.txt for a full report.
112 verbose exit 1

**Comment by fzaninotto** [comment]
Hi�,

your console log isn't readable�����

Also, we don't support Windows systems as we don't develop on it. 

Finally, ����� is ���  but ������� � with ��� ����� of course.

**Comment by BitBoxBear** [comment]
`verbose stack Error: unable to resolve dependency tree`
Correct me if I wrong but resolving of dependency tree is not a windows issue. Its a package dependency issue.
```
0 verbose cli [
0 verbose cli   'D:\\NodeJS\\node.exe',
0 verbose cli   'D:\\NodeJS\\node_modules\\npm\\bin\\npm-cli.js',
0 verbose cli   'install',
0 verbose cli   'react-admin'
0 verbose cli ]
1 info using npm@7.5.1
2 info using node@v15.8.0
3 timing config:load:defaults Completed in 0ms
4 timing config:load:file:D:\NodeJS\node_modules\npm\npmrc Completed in 1ms
5 timing config:load:builtin Completed in 2ms
6 timing config:load:cli Completed in 0ms
7 timing config:load:env Completed in 1ms
8 timing config:load:file:E:\test-admin\.npmrc Completed in 0ms
9 timing config:load:project Completed in 0ms
10 timing config:load:file:C:\Users\X\.npmrc Completed in 1ms
11 timing config:load:user Completed in 1ms
12 timing config:load:file:C:\Users\X\AppData\Roaming\npm\etc\npmrc Completed in 0ms
13 timing config:load:global Completed in 0ms
14 timing config:load:cafile Completed in 0ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:setUserAgent Completed in 0ms
17 timing config:load:setEnvs Completed in 1ms
18 timing config:load Completed in 5ms
19 verbose npm-session cfee11eb225ae5e4
20 timing npm:load Completed in 11ms
21 timing arborist:ctor Completed in 1ms
22 timing idealTree:init Completed in 461ms
23 timing idealTree:userRequests Completed in 1ms
24 silly idealTree buildDeps
25 silly fetch manifest react-admin@*
26 http fetch GET 200 https://registry.npmjs.org/react-admin 19ms (from cache)
27 silly fetch manifest react@^17.0.1
28 http fetch GET 304 https://registry.npmjs.org/react 616ms (from cache)
29 silly fetch manifest react-dom@^17.0.1
30 http fetch GET 200 https://registry.npmjs.org/react-dom 2ms (from cache)
31 silly placeDep ROOT react-admin@3.12.4 OK for: test-admin@0.1.0 want: *
32 silly fetch manifest @material-ui/core@^4.11.2
33 silly fetch manifest @material-ui/icons@^4.11.2
34 silly fetch manifest @material-ui/styles@^4.11.2
35 silly fetch manifest connected-react-router@^6.5.2
36 silly fetch manifest final-form@^4.20.0
37 silly fetch manifest final-form-arrays@^3.0.1
38 silly fetch manifest ra-core@^3.12.4
39 silly fetch manifest ra-i18n-polyglot@^3.12.4
40 silly fetch manifest ra-language-english@^3.12.4
41 silly fetch manifest ra-ui-materialui@^3.12.4
42 silly fetch manifest react-final-form@^6.3.3
43 silly fetch manifest react-final-form-arrays@^3.1.1
44 silly fetch manifest react-redux@^7.1.0
45 silly fetch manifest react-router@^5.1.0
46 silly fetch manifest react-router-dom@^5.1.0
47 silly fetch manifest redux@^3.7.2 || ^4.0.3
48 silly fetch manifest redux-saga@^1.0.0
49 http fetch GET 200 https://registry.npmjs.org/connected-react-router 34ms (from cache)
50 http fetch GET 200 https://registry.npmjs.org/final-form-arrays 34ms (from cache)
51 http fetch GET 200 https://registry.npmjs.org/ra-core 34ms (from cache)
52 http fetch GET 200 https://registry.npmjs.org/final-form 37ms (from cache)
53 http fetch GET 200 https://registry.npmjs.org/ra-i18n-polyglot 36ms (from cache)
54 http fetch GET 200 https://registry.npmjs.org/react-final-form-arrays 35ms (from cache)
55 http fetch GET 200 https://registry.npmjs.org/react-final-form 36ms (from cache)
56 http fetch GET 200 https://registry.npmjs.org/ra-ui-materialui 39ms (from cache)
57 http fetch GET 200 https://registry.npmjs.org/react-router 38ms (from cache)
58 http fetch GET 200 https://registry.npmjs.org/ra-language-english 42ms (from cache)
59 http fetch GET 304 https://registry.npmjs.org/@material-ui%2fstyles 99ms (from cache)
60 http fetch GET 304 https://registry.npmjs.org/redux-saga 113ms (from cache)
61 http fetch GET 304 https://registry.npmjs.org/redux 116ms (from cache)
62 http fetch GET 304 https://registry.npmjs.org/react-redux 123ms (from cache)
63 http fetch GET 304 https://registry.npmjs.org/react-router-dom 127ms (from cache)
64 http fetch GET 304 https://registry.npmjs.org/@material-ui%2fcore 162ms (from cache)
65 http fetch GET 304 https://registry.npmjs.org/@material-ui%2ficons 168ms (from cache)
66 timing idealTree:#root Completed in 819ms
67 silly fetch manifest @types/react@^16.8.6 || ^17.0.0
68 http fetch GET 304 https://registry.npmjs.org/@types%2freact 55ms (from cache)
69 silly fetch manifest react@^16.9.0 || ^17.0.0
70 silly fetch manifest react-dom@^16.9.0 || ^17.0.0
71 silly fetch manifest history@^4.7.2
72 http fetch GET 304 https://registry.npmjs.org/history 54ms (from cache)
73 silly fetch manifest immutable@^3.8.1 || ^4.0.0-rc.1
74 http fetch GET 304 https://registry.npmjs.org/immutable 67ms (from cache)
75 silly fetch manifest react@^16.4.0
76 silly fetch manifest seamless-immutable@^7.1.3
77 http fetch GET 200 https://registry.npmjs.org/seamless-immutable 2ms (from cache)
78 silly placeDep ROOT @material-ui/core@4.11.3 OK for: react-admin@3.12.4 want: ^4.11.2
79 silly placeDep ROOT @material-ui/icons@4.11.2 OK for: react-admin@3.12.4 want: ^4.11.2
80 silly placeDep ROOT @material-ui/styles@4.11.3 OK for: react-admin@3.12.4 want: ^4.11.2
81 silly placeDep node_modules/react-admin connected-react-router@6.8.0 OK for: react-admin@3.12.4 want: ^6.5.2
82 silly placeDep ROOT immutable@4.0.0-rc.12 OK for: connected-react-router@6.8.0 want: ^3.8.1 || ^4.0.0-rc.1
83 silly placeDep ROOT history@4.10.1 OK for: connected-react-router@6.8.0 want: ^4.7.2
84 timing idealTree Completed in 1490ms
85 timing command:install Completed in 1494ms
86 verbose stack Error: unable to resolve dependency tree
86 verbose stack     at Arborist.[failPeerConflict] (D:\NodeJS\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\build-ideal-tree.js:1115:25)
86 verbose stack     at Arborist.[placeDep] (D:\NodeJS\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\build-ideal-tree.js:1201:30)
86 verbose stack     at Arborist.[placeDep] (D:\NodeJS\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\build-ideal-tree.js:1339:41)
86 verbose stack     at D:\NodeJS\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\build-ideal-tree.js:835:46
86 verbose stack     at Array.map (<anonymous>)
86 verbose stack     at Arborist.[buildDepStep] (D:\NodeJS\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\build-ideal-tree.js:835:8)
86 verbose stack     at async Arborist.buildIdealTree (D:\NodeJS\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\build-ideal-tree.js:209:7)
86 verbose stack     at async Promise.all (index 1)
86 verbose stack     at async Arborist.reify (D:\NodeJS\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\reify.js:127:5)
86 verbose stack     at async install (D:\NodeJS\node_modules\npm\lib\install.js:38:3)
87 verbose cwd E:\test-admin
88 verbose Windows_NT 10.0.19042
89 verbose argv "D:\\NodeJS\\node.exe" "D:\\NodeJS\\node_modules\\npm\\bin\\npm-cli.js" "install" "react-admin"
90 verbose node v15.8.0
91 verbose npm  v7.5.1
92 error code ERESOLVE
93 error ERESOLVE unable to resolve dependency tree
94 error
95 error Found: react@17.0.1
95 error node_modules/react
95 error   peer react@"*" from @testing-library/react@11.2.5
95 error   node_modules/@testing-library/react
95 error     @testing-library/react@"^11.2.5" from the root project
95 error   peer react@"17.0.1" from react-dom@17.0.1
95 error   node_modules/react-dom
95 error     peer react-dom@"*" from @testing-library/react@11.2.5
95 error     node_modules/@testing-library/react
95 error       @testing-library/react@"^11.2.5" from the root project
95 error     react-dom@"^17.0.1" from the root project
95 error     4 more (react-admin, @material-ui/core, @material-ui/icons, @material-ui/styles)
95 error   6 more (react-scripts, the root project, react-admin, ...)
95 error
95 error Could not resolve dependency:
95 error peer react@"^16.4.0" from connected-react-router@6.8.0
95 error node_modules/react-admin/node_modules/connected-react-router
95 error   connected-react-router@"^6.5.2" from react-admin@3.12.4
95 error   node_modules/react-admin
95 error     react-admin@"*" from the root project
95 error
95 error Fix the upstream dependency conflict, or retry
95 error this command with --force, or --legacy-peer-deps
95 error to accept an incorrect (and potentially broken) dependency resolution.
95 error
95 error See C:\Users\X\AppData\Local\npm-cache\eresolve-report.txt for a full report.
96 verbose exit 1
```

But hey I was just trying to be nice and let you know this is a problem. If you look at line 109 in the 1st post, I took the time to highlight it for you, it should not matter if its windows or other OS. Its an NPM dependency package issue.

��� ����� of course.

**Comment by fzaninotto** [comment]
Tested on Linux: no problem

```
➜  npm -v
6.14.4
➜  node -v
v12.16.1
➜  npm install react-admin
[...]
+ react-admin@3.12.4
added 153 packages from 186 contributors and audited 153 packages in 49.267s
```

So either it's a problem with your npm cache, or with a more recent version of node/npm, or with Windows.

**Comment by fzaninotto** [comment]
Reproduced with node v15.8.0 on Linux

```
➜  tmp nvm use v15.8.0
Now using node v15.8.0 (npm v7.5.1)
➜  tmp npm install react-admin
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! Found: react@17.0.1
npm ERR! node_modules/react
npm ERR!   peer react@"^16.9.0 || ^17.0.0" from react-admin@3.12.4
npm ERR!   node_modules/react-admin
npm ERR!     react-admin@"*" from the root project
npm ERR!   peer react@"17.0.1" from react-dom@17.0.1
npm ERR!   node_modules/react-dom
npm ERR!     peer react-dom@"^16.9.0 || ^17.0.0" from react-admin@3.12.4
npm ERR!     node_modules/react-admin
npm ERR!       react-admin@"*" from the root project
npm ERR!     peer react-dom@"^16.8.0 || ^17.0.0" from @material-ui/core@4.11.3
npm ERR!     node_modules/@material-ui/core
npm ERR!       @material-ui/core@"^4.11.2" from react-admin@3.12.4
npm ERR!       node_modules/react-admin
npm ERR!         react-admin@"*" from the root project
npm ERR!       1 more (@material-ui/icons)
npm ERR!     2 more (@material-ui/icons, @material-ui/styles)
npm ERR!   3 more (@material-ui/core, @material-ui/icons, @material-ui/styles)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.4.0" from connected-react-router@6.8.0
npm ERR! node_modules/react-admin/node_modules/connected-react-router
npm ERR!   connected-react-router@"^6.5.2" from react-admin@3.12.4
npm ERR!   node_modules/react-admin
npm ERR!     react-admin@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See /home/francois/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/francois/.npm/_logs/2021-02-17T18_20_06_923Z-debug.log
```

**Comment by muhadbk0** [comment]
Apples-Air:admin technewwings$ npm install --save-dev react-admin
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! Found: react@17.0.1
npm ERR! node_modules/react
npm ERR!   peer react@"^16.9.0 || ^17.0.0" from react-admin@3.12.4
npm ERR!   node_modules/react-admin
npm ERR!     dev react-admin@"*" from the root project
npm ERR!   peer react@"17.0.1" from react-dom@17.0.1
npm ERR!   node_modules/react-dom
npm ERR!     peer react-dom@"^16.9.0 || ^17.0.0" from react-admin@3.12.4
npm ERR!     node_modules/react-admin
npm ERR!       dev react-admin@"*" from the root project
npm ERR!     peer react-dom@"^16.8.0 || ^17.0.0" from @material-ui/core@4.11.3
npm ERR!     node_modules/@material-ui/core
npm ERR!       @material-ui/core@"^4.11.2" from react-admin@3.12.4
npm ERR!       node_modules/react-admin
npm ERR!         dev react-admin@"*" from the root project
npm ERR!       1 more (@material-ui/icons)
npm ERR!     2 more (@material-ui/icons, @material-ui/styles)
npm ERR!   3 more (@material-ui/core, @material-ui/icons, @material-ui/styles)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.4.0" from connected-react-router@6.8.0
npm ERR! node_modules/react-admin/node_modules/connected-react-router
npm ERR!   connected-react-router@"^6.5.2" from react-admin@3.12.4
npm ERR!   node_modules/react-admin
npm ERR!     dev react-admin@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/technewwings/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/technewwings/.npm/_logs/2021-02-17T18_49_52_337Z-debug.log
Apples-Air:admin technewwings$ node -v
v14.15.3

**Comment by fzaninotto** [comment]
This is caused by npm 7.x, which broke BC by installing peer dependences (https://github.blog/2020-10-13-presenting-v7-0-0-of-the-npm-cli/). The connected-react-router package must be modified to allow react 17 as peer dependency. 

As this is an external package, we can't really solve this problem in react-admin.

In the meantime, we recommend that you use yarn or downgrade to npm 6.

**Comment by fzaninotto** [comment]
Fixed by `connected-react-router` v6.9.1.

**Comment by dylanh724** [comment]
> Fixed by `connected-react-router` v6.9.1.

Wait, @fzaninotto apologies, what do you mean fixed "by `connected-react-router`? How can I resolve this?
